lib/vssh/libssh.c

```diff
@@ -1965,8 +1965,7 @@ static CURLcode myssh_statemach_act(struct Curl_easy *data, bool *block)
         /* conn->sock[FIRSTSOCKET] is closed by ssh_disconnect behind our back,
            explicitly mark it as closed with the memdebug macro. This libssh
            bug is fixed in 0.10.0. */
-        fake_sclose(conn->sock[FIRSTSOCKET]);
-        conn->sock[FIRSTSOCKET] = CURL_SOCKET_BAD;
+        Curl_conn_forget_socket(data, FIRSTSOCKET);
```

Comment:
should the comment just before this, that explains the use of **fake_** sclose here not also be carried over to the new function? Otherwise it is hard to follow...

Reply:
You want me to change even more lines? Hmm...okay then!